app/scripts/controllers/transactions/tx-state-manager.test.js

```diff
@@ -540,72 +573,56 @@ describe('TransactionStateManager', function () {
 
     it('cuts off early txs beyond a limit', function () {
       const limit = txStateManager.txHistoryLimit;
-      for (let i = 0; i < limit + 1; i++) {
-        const tx = {
-          id: i,
-          time: new Date(),
-          status: TRANSACTION_STATUSES.CONFIRMED,
-          metamaskNetworkId: currentNetworkId,
-          txParams: {
-            to: VALID_ADDRESS,
-            from: VALID_ADDRESS,
-          },
-        };
-        txStateManager.addTransaction(tx);
-      }
+      const txs = generateTransactions(limit + 1, {
+        chainId: currentChainId,
+        to: VALID_ADDRESS,
+        from: VALID_ADDRESS_TWO,
+        status: TRANSACTION_STATUSES.CONFIRMED,
+      });
+      txs.forEach((tx) => txStateManager.addTransaction(tx));
       const result = txStateManager.getTransactions();
       assert.equal(result.length, limit, `limit of ${limit} txs enforced`);
       assert.equal(result[0].id, 1, 'early txs truncated');
     });
 
     it('cuts off early txs beyond a limit whether or not it is confirmed or rejected', function () {
       const limit = txStateManager.txHistoryLimit;
-      for (let i = 0; i < limit + 1; i++) {
-        const tx = {
-          id: i,
-          time: new Date(),
-          status: TRANSACTION_STATUSES.REJECTED,
-          metamaskNetworkId: currentNetworkId,
-          txParams: {
-            to: VALID_ADDRESS,
-            from: VALID_ADDRESS,
-          },
-        };
-        txStateManager.addTransaction(tx);
-      }
+      const txs = generateTransactions(limit + 1, {
+        chainId: currentChainId,
+        to: VALID_ADDRESS,
+        from: VALID_ADDRESS_TWO,
+        status: TRANSACTION_STATUSES.CONFIRMED,
```

Comment:
Given the description of this test and that the previous one is identical aside from the status, I think the intention was to ensure that it worked the same with the status was `REJECTED` as it did when it was `CONFIRMED`

```suggestion
        status: TRANSACTION_STATUSES.REJECTED,
```